src/main/java/com/kodilla/ecommercee/scheduler/EmailScheduler.java

```diff
@@ -0,0 +1,36 @@
+package com.kodilla.ecommercee.scheduler;
+
+import com.kodilla.ecommercee.config.AdminConfig;
+import com.kodilla.ecommercee.entity.Mail;
+import com.kodilla.ecommercee.entity.User;
+import com.kodilla.ecommercee.entity.UserOrder;
+import com.kodilla.ecommercee.repository.OrderRepository;
+import com.kodilla.ecommercee.service.EmailService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.util.List;
+
+public class EmailScheduler {
+
+    private static final String SUBJECT ="Email: New order list:";
+
+    @Autowired
+    private EmailService emailService;
+
+    @Autowired
+    private AdminConfig adminConfig;
+
+    @Autowired
+    private OrderRepository orderRepository;
+
+    @Scheduled(cron ="0 0/30 * * * *")
```

Comment:
Jest ok, w tasku było zastosowanie Schedullera, ale można by pomyśleć na przyszłość, żeby wysyłał maila w momencie zatwierdzenia zamówienia, czyli jak UserOrder->resolved zmieni się na true, czyli w OrderController,

Reply:
Tutaj mam problem, wydaje mi się że powinna być metoda która zmieni w bazie danych UserOrder->  mailSent ->true; Nie wiem za bardzo jak to zrobić?

Reply:
w pętli niżej iterujesz po List i masz każdego userOrder -> robisz userOrder.setMailSent(true), a potem userRepository.save(userOrder). Wstrzyknij tylko przez @Autowired UserRepository userRepository. Trzeba by tylko wykombinować coś, żeby w razie błędu w wysyłce maila nie zmieniał jednak tego pola na true. Najprościej byłoby nieco przerobić metode send tak, żeby zwracała true, jak wszystko jest ok. Pamiętaj tez o pojedynczej odpowiedzialności metod -> zapis do bazy powienie być w osobnej metodzie,

Reply:
Hmm ... Jeżeli dobrze rozumiem, to średnio się z tym zgadzam, ponieważ UserRepository definiuje użytkownika. OrderRepository określa zlecenie. Jeden użytkownik może mieć wiele zleceń, a sklep ma wiele użytkowników w bazie. No to chyba zmienna logiczna czy mail został wysłany powinna odnosić się do zlecenia a nie do użytkownika.

Reply:
I masz rację, wieczorem już zmęczony bywam (tak jak kiedyś pisałeś myli się UserOrder z User i Order). Beana OrderRepository już masz, nic nie musisz wstrzykiwać. W pętli dodaj po prostu zapis userOrder ze zmienionym polem mailSent. Warto by dodać warunek, że jeśli z wysłaniem maila poszło coś nie tak (błąd), to pole mailSent powinno pozostać nie zmienione (false)